WIKI.org

```diff
@@ -78,17 +80,15 @@ to your Dropbox, here are your options:
  #+BEGIN_EXAMPLE
  Alias /webdav /srv/dav/
 
- RewriteEngine On
- RewriteCond %{REQUEST_METHOD} OPTIONS
- RewriteRule ^(.*)$ $1 [R=200,L]
-
  <Location /webdav>
      Options Indexes
      DAV On
      AuthType Basic
      AuthName "webdav"
      AuthUserFile /srv/dav/.htpasswd
-     Require valid-user
+     <LimitExcept OPTIONS>
```

Comment:
It's great that you're putting a lot of effort into better docs and setup for WebDAV. It'll certainly be very much appreciated by future readers!

At this time, I cannot test this new configuration for Apache, though. When I wrote the original docs, I did set this up locally, so that I could test WebDAV. Since then, I have switched computers and haven't set this up, again.

I'm hesitant to merge config that I haven't checked. By this, I don't want to imply that it's not working for you, but as a maintainer I should be able to understand/reproduce the setup.

The best way to move forward would likely be to create this Apache setup in a `Dockerfile`. I've been thinking of doing that for some time now. It would ease a lot of discussions on the topic of WebDAV. Right now, everyone is using some very specific setup (NextCloud, HA Proxy, Apache, etc) and it's not really possible to give support for that. Such Ops topics are way out of scope of a SPA project. If we had an official example Docker container that works for us and we use in development, we can always point to that one.

What do you think? Maybe you're even already using a Docker based setup?(;

Reply:
I'm not using Docker but this is an excellent idea.

Totally understand your hesitation.  I just spent a frustrating 3-4 hours trying to really get to the bottom of this Apache voodoo.  I conducted countless experiments, and I'm now pretty confident `LimitExcept` is the better way to go.  It has clear logic to it (prevent a CORS preflight `OPTIONS` check from resulting in 401 response due to lack of credentials) whereas the "use `RewriteRule ^(.*)$ $1 [R=200,L]` to rewrite a URL to itself" trick is much weirder and hard to explain.  In fact, can you explain that?  If so, I'd love to hear the explanation so I can fully understand it.  If not then you have a choice between two configs you don't fully understand ;-)

BTW it makes no sense to me to have the rewrite rules applying more generally, outside the DAV `<Directory>` or `<Location>` section, because we only want CORS checks to succeed for DAV.

Reply:
@aspiers Very well, then. I've just spend some time and have setup such a Docker based setup and rewrote the documentation accordingly: https://github.com/200ok-ch/organice/pull/356

If you generally approve of the PR, I'll merge it and then I can add your proposed changes to the official test bench and see if it still works on my machine(;

NB: If you can get the authentication to work using the Docker image, I don't mind if you enable it and propose a change. I tried quite a few times, but failed. Since it's just a test bench, it can live without authentication.

Reply:
#356 looks good to me, please merge and then we can resume progress on this PR.  I'd be interested to hear what blockers you hit regarding authentication.  I wonder if this PR helps with any of them.